packages/@lwc/template-compiler/src/parser/expression.ts

```diff
@@ -119,7 +118,14 @@ export function parseIdentifier(
     source: string,
     location: Location
 ): TemplateIdentifier {
-    if (esutils.keyword.isIdentifierES6(source)) {
+    let isValid = true;
+
+    isValid = isIdentifierStart(source.charCodeAt(0));
+    for (let i = 1; i < source.length && isValid; i++) {
+        isValid = isIdentifierChar(source.charCodeAt(i));
```

Comment:
Am I misreading this, or will `isValid` actually always be true as long as the last character is a valid character?

Reply:
Correct. `isValid` is always true either the first character or the current character set it to false. In this case, the loop stops. I am open to making any adjustments to clarify this.

I also added a couple of test cases to prevent future regression.

Reply:
Ah, I missed the `&& isValid`. My mistake.

If it were me, I probably would have done `if (!isValid) { break; }` but your version is much more concise.